client/src/api/datasets.ts

```diff
@@ -49,25 +49,21 @@ export async function fetchDatasetDetails(params: { id: string }): Promise<Datas
     return data as unknown as DatasetDetails;
 }
 
-const updateHistoryDataset = fetcher.path("/api/histories/{history_id}/contents/{type}s/{id}").method("put").create();
+const updateDataset = fetcher.path("/api/datasets/{dataset_id}").method("put").create();
 
-export async function undeleteHistoryDataset(historyId: string, datasetId: string) {
-    const { data } = await updateHistoryDataset({
-        history_id: historyId,
-        id: datasetId,
+export async function undeleteHistoryDataset(datasetId: string) {
```

Comment:
I would rename this to `undeleteDataset` again for consistency, but that can be done in https://github.com/galaxyproject/galaxy/pull/17853. So no need to rebase, etc.